spec/models/vote_spec.rb

```diff
@@ -16,81 +16,172 @@
     it "receives vote operation from user" do
       allow(Vote).to receive(:process_vote).and_return (nil)
       expect(Vote.act_on_vote(operation, subject, subject_id, user)).to be nil
-     end
+    end
   end
 
   describe ".process_vote" do
-   it "returns an integer value after vote is processed" do
-    allow(Vote).to receive(:subject_exists?).and_return (true)
-    allow(Vote).to receive(:store_vote).and_return (true)
-    expect(Vote.process_vote(subject_class, subject_id, user, value)).to be_an Integer
-    expect((Vote.process_vote(subject_class, subject_id, user, value).abs)).to eq(1)
-   end
+    it "returns an integer value after vote is processed" do
+      allow(Vote).to receive(:subject_exists?).and_return (true)
+      allow(Vote).to receive(:store_vote).and_return (7)
+      expect(Vote.process_vote(subject_class, subject_id, user, value)).to be_an Integer
+      expect((Vote.process_vote(subject_class, subject_id, user, value).abs)).to eq(1)
+    end
   end
 
   describe ".process_vote" do
-   it "returns nil" do
-    allow(Vote).to receive(:subject_exists?).and_return (nil)
-    expect(Vote.process_vote(subject_class, subject_id, user, value)).to be nil
-   end
+    it "returns nil" do
```

Comment:
Why not merge the two `describe ".process_vote"` as such all their expectations will be together and easily managed.